### ISSUE TITLE
Clothing layers

### ISSUE THREAD

**Issue by DcNdrew** [Enginecrafter77/SurvivalInc#6]
There should be another layer for clothes under the armor so you can add socks, trousers, shirts and headgear.

**Comment by Enginecrafter77** [comment]
This one is going to be especially tricky. Like, I could possibly made it that you can pad your existing armor for isolation, but it would just double the amount of registered armor in the game. The (optimal) solution would be to add new slots to the player inventory, which is the tricky part. I have only seen 2 mods done this in the past: Tinker's Construct and Galacticraft.

**Comment by hazerddex** [comment]
an easier solution is to just add compatibility with  https://www.curseforge.com/minecraft/mc-mods/armor-underwear-mod/files

**Comment by Enginecrafter77** [comment]
That sounds interesting. However, as you said, the mod is probably no logner being updated or actively developed. But despite that, it would be much easier and also significantly less error prone to use something already available at hand, rather than composing a whole new system. Anyways, it was meant for TAN, so I think it fits the idea nicely.

**Comment by mxnmnm** [comment]
Would it be possible to make an addon to baubles which would add such slots? I know Zabi made a thing which added more rings https://www.curseforge.com/minecraft/mc-mods/bring-me-the-rings
I don't know how difficult it would be to add new types of slots but I figured it may be easier using baubles as a backbone. There is also cosmetic armor which also adds different slots. Just different places you can start looking if you need inspiration lol.

**Comment by Enginecrafter77** [comment]
@mxnmnm The Baubles approach seems kind of promising to me. I have never looked into it that much, so maybe it's the right time to do so. It seems Zabi actually made a coremod and used ASM to accomplish adding additional slots. But I guess using the pre-existing baubles slots is still an option. Maybe only that it may conflict with other mods, if you know what I mean.

**Comment by mxnmnm** [comment]
Maybe it should use ring slots so you can combine it with bring me the rings lol

**Comment by Enginecrafter77** [comment]
I thought about using the 4 rightmost bauble slots. Do you think it would be very intrusive regarding other mods that use these slots? I mean, like, are these slots used for something besides thaumcraft's goggles of revealing? (the only item I can think of atm)

**Comment by mxnmnm** [comment]
Yea it would be a bad idea I can give you a list of the mods i use (over 300 so a pretty good idea of a worstcase scenario) that would conflict with it

**Comment by mxnmnm** [comment]
An alternative which is what was originally suggested. I know a mod that has a similar thing. The betweenlands adds gems which you can add onto any armor. I'm guessing the whole adding extra data to an item of a specific type is the main issue here. So that might make it easier for you to implement something similar.

**Comment by mxnmnm** [comment]
If baubles compat was the route chosen, I would recommend you make it possible for any slot just for simplicity's sake.

**Comment by mxnmnm** [comment]
Is there a discord where we can talk about this so we can talk together a bit more smoothly?

**Comment by Enginecrafter77** [comment]
Well, there wasn't until now :D. https://discord.gg/bHuzSB

**Comment by zabi94** [comment]
I have no idea what this is, but why not just make a completely separate inventory with new slots, then add a keybinding to open it, and/or overlay a button on the inventory screen that just opens a new screen?
Maximum compatibility and control, easy to implement (seriously, don't go the bring me the rings way, it was hell to implement, it's raw asm), no need to have baubles as a dependency.

**Comment by Enginecrafter77** [comment]
@zabi94 Yeah we discussed it on discord, and we settled on the separate inventory approach.
